Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/GatewayService.java

```diff
@@ -0,0 +1,266 @@
+ *      * Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following disclaimer.
+ *      * Redistributions in binary form must reproduce the above copyright
+ *        notice, this list of conditions and the following disclaimer in the documentation
+ *        and/or other materials provided with the distribution.
+ *      * Neither the name of the United States Government nor the
+ *        names of its contributors may be used to endorse or promote products
+ *        derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ *  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ *  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.admingui.services;
+
+import gov.hhs.fha.nhinc.admingui.event.model.Document;
+import gov.hhs.fha.nhinc.admingui.event.model.Patient;
+import gov.hhs.fha.nhinc.admingui.managed.PatientSearchBean;
+import gov.hhs.fha.nhinc.admingui.services.exception.DocumentMetadataException;
+import gov.hhs.fha.nhinc.admingui.services.impl.DocumentQueryServiceImpl;
+import gov.hhs.fha.nhinc.admingui.services.impl.DocumentRetrieveServiceImpl;
+import gov.hhs.fha.nhinc.admingui.services.impl.PatientServiceImpl;
+import gov.hhs.fha.nhinc.docquery.builder.impl.FindDocumentsAdhocQueryRequestBuilder;
+import gov.hhs.fha.nhinc.docquery.model.DocumentMetadata;
+import gov.hhs.fha.nhinc.docquery.model.DocumentMetadataResult;
+import gov.hhs.fha.nhinc.docquery.model.DocumentMetadataResults;
+import gov.hhs.fha.nhinc.docquery.model.builder.impl.DocumentMetadataResultsModelBuilderImpl;
+import gov.hhs.fha.nhinc.docretrieve.model.DocumentRetrieve;
+import gov.hhs.fha.nhinc.docretrieve.model.DocumentRetrieveResults;
+import gov.hhs.fha.nhinc.patientdiscovery.model.PatientSearchResults;
+import gov.hhs.fha.nhinc.util.format.UTCDateUtil;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import org.apache.log4j.Logger;
+
+/**
+ * Singleton Interface class between UI and the backend services. provides high level APIs for calling PD, DQ, RD etc.
+ *
+ *
+ * @author Naresh Subramanyan
+ */
+public class GatewayService {
+
+    private static final Logger LOG = Logger.getLogger(GatewayService.class);
+
+    private PatientService patientService;
```

Comment:
these services should all be autowired.  should follow our same injection strategy that we use throughout AdminGUI.